src/osd/modules/render/bgfx/shadermanager.cpp

```diff
@@ -83,6 +83,15 @@ std::string shader_manager::make_path_string(const osd_options &options, const s
 {
 	std::string shader_path(options.bgfx_path());
 	shader_path += PATH_SEPARATOR "shaders" PATH_SEPARATOR;
+
+#if defined(SDLMAME_EMSCRIPTEN)
+	// Hard-code renderer type to OpenGL ES for emscripten builds since the
+	// bgfx::getRendererType() is called here before BGFX has been
+	// initialized and therefore gives the wrong renderer type (Noop).
+	shader_path += "essl" PATH_SEPARATOR;
+	return shader_path;
+#endif
+
```

Comment:
Why doesn’t it also need a hack on macOS or Linux where DirectX 9 is unsupported?  Shouldn’t bgfx be equally uninitialised on those platforms?

Reply:
I’m not sure why the behavior differs, but this could be related to platform-specific handling within the bgfx codebase or how certain APIs are abstracted across different systems. The bgfx docs states that the library (bgfx) must be initialised before calling the function. As it is now, mame will call the function before bgfx has been initialised and for emscripten the wrong rendering type is returned. After bgfx has been initialised, the function returns the correct type for emscripten.